hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -144,12 +146,13 @@ private void enrichHostHeader(Event event) {
     }
   }
 
-  private Optional<String> getGrpcAuthority(Event event) {
+  private Optional<String> getGrpcHostHeader(Event event) {
     Optional<String> grpcAuthority = RpcSemanticConventionUtils.getGrpcAuthority(event);
     if (grpcAuthority.isPresent()) {
       return getSanitizedHostValue(grpcAuthority.get());
+    } else {
+      return getGrpcRequestMetadataHost(event);
```

Comment:
Look for the method - `getGrpcXForwardedFor`, this method should be similar to that. It should internally check that isRpcSystemGrpc, if yes, it should check for the `rpc.request.metadata.host` attribute.

Reply:
Second, we also have to sanitize the host value read from other attributes - `getSanitizedHostValue (getGrpcRequestMetadataHost())`

Reply:
resolved